### ISSUE TITLE
Pyright report bad type hints in a non-annotation context

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#1067]
**Describe the bug**

Pyright is reporting bad type hint that appears as run time value instead of annotation.

Type hints can be used in runtime for runtime type check (e.g. [typeguard](https://typeguard.readthedocs.io/en/latest/index.html)). Somehow Pyright picked up those runtime type hints and report errors for unsupported features.

For example:

``` python
    import typeguard
    from typing import Literal

    obj = object()
    typeguard.check_type("obj", obj, Literal[obj])                 
```
Pyright reports error `Type arguments for "Literal" must be None, a literal value (int, bool, str, or bytes), or an enum value` for `Literal[obj]` but this is a valid case to use `Literal` and typeguard library and has nothing to do with static type checker.

**Expected behavior**
Pyright should ignore type hints that do not appeared in annotation.

**VS Code extension or command-line**
vscode extension v1.1.75

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.76, which I just published. It will also be included in the next release of Pylance.
